ec2.tf

```diff
@@ -0,0 +1,120 @@
+data "aws_ssm_parameter" "ami" {
+  name = "/aws/service/ecs/optimized-ami/amazon-linux-2/recommended/image_id"
+}
+
+locals {
+  ec2_capacity_providers = local.enabled ? var.capacity_providers_ec2 : {}
+
+  instance_profile_name = join("", aws_iam_instance_profile.default.*.name)
+}
+
+locals {
+  user_data = {
+    for name, provider in local.ec2_capacity_providers :
+    name => <<EOT
+#!/bin/bash
+echo ECS_CLUSTER="${local.cluster_name}" >> /etc/ecs/ecs.config
+echo ECS_ENABLE_CONTAINER_METADATA=true >> /etc/ecs/ecs.config
+echo ECS_POLL_METRICS=true >> /etc/ecs/ecs.config
+echo ECS_ENABLE_SPOT_INSTANCE_DRAINING=${provider.instance_market_options != null && provider.mixed_instances_policy != null} >> /etc/ecs/ecs.config
+echo ECS_WARM_POOLS_CHECK=${provider.warm_pool != null} >> /etc/ecs/ecs.config
+
+${replace(provider.user_data, "#!/bin/bash", "")}
+EOT
+  }
+}
+
+module "ecs_labels" {
+  for_each = local.ec2_capacity_providers
+  source   = "cloudposse/label/null"
+  version  = "0.25.0" # requires Terraform >= 0.13.0
+
+  enabled    = var.enabled
+  attributes = concat(module.this.context.attributes, [each.key])
+  tags       = merge(module.this.context.tags, { "AmazonECSManaged" : "true" })
+  context    = module.this.context
+}
+
+output "test" {
+  value = var.capacity_providers_ec2
+}
+
+module "autoscale_group" {
+  for_each = local.ec2_capacity_providers
```

Comment:
The asg is not needed if using a fargate cluster.

Reply:
there is variable `capacity_providers_fargate` flag, `ec2_capacity_providers` could be empty for that case.